processing/src/main/java/org/apache/druid/frame/key/ByteRowKeyComparator.java

```diff
@@ -44,23 +45,30 @@ public class ByteRowKeyComparator implements Comparator<byte[]>
   private final int firstFieldPosition;
   private final RowKeyComparisonRunLengths rowKeyComparisonRunLengths;
   private final RowSignature rowSignature;
+  private final ComplexMetricSerde[] complexMetricSerdes;
```

Comment:
Could use a couple of things to aid understanding:

1. Comment explaining that the index here is the index in `rowKeyComparisonRunLengths`, and that it's null if there are no serdes.
2. `@Nullable` annotation.

Reply:
Regarding 2, the array is not null if there are no serdes, it's filled with nulls.